hooks/useStorageBucket.ts

```diff
@@ -0,0 +1,53 @@
+import axios from "axios"
+import { TicketsMetadata } from "../dtos/ticket/metadata.dto"
+import { StoredAsset } from "../interfaces/gcp/storage.interface"
+
+const URL = process.env.NEXT_PUBLIC_WARDEN_API_URL
+
+export function useStorageBucket() {
+  async function saveFile(
+    files: File[],
+    pathName: string,
+    metadata?: TicketsMetadata[],
+    onUploadProgress?: () => void
+  ): Promise<StoredAsset[]> {
+    const formData = new FormData()
+    files.forEach((file) => {
+      formData.append("files", file)
+    })
+    formData.append("metadata", JSON.stringify(metadata ?? []))
+    formData.append("folder", pathName)
+    const res = await axios.post<StoredAsset[]>(`${URL}/media`, formData, {
+      headers: {
+        "Content-Type": "multipart/form-data"
+      },
+      onUploadProgress
```

Comment:
what if `onUploadProgress` is undefined?

Reply:
It can be undefined. As I have tested, no error occurs